src/eth_provider/error.rs

```diff
@@ -221,6 +225,14 @@ pub enum ReceiptError {
     ConversionError,
 }
 
+/// Error related to headers.
+#[derive(Debug, Error)]
+pub enum HeaderError {
+    /// Error related to conversion.
+    #[error("conversion error")]
+    ConversionError,
+}
+
```

Comment:
Should we maybe have a `EthereumDataFormatError` or `EthereumTypeCastError` and have all the HeaderError, TransactionConversionError,... there?

Reply:
Done, in `KakarotError` I left the generic conversion error for specific needs. Otherwise I encapsulated the header, receipt and transaction conversion errors in a new error enum, it's cleaner that way, especially for future needs. We can control the granularity of the errors that we want to put with possibly subtypes.

Reply:
That looks good, but I think we can remove the generic Conversion Error, it's actually only used for conversion error between Eth and Starknet types, which could just be a `EthereumDataFormatError`. But this can be done in another PR, I will make an issue